### PR TITLE
Started integration testing implementing

### PR DESCRIPTION
**Pull request by cyberdr0id**
Database integration tests

### CODE REVIEW

File: tests/main_test.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+func (s *ReferralAPISuite) TearDownSuite() {
+	if err := s.db.Close(); err != nil {
+		s.FailNow(fmt.Errorf("cannot close database connection: %w", err).Error())
+	}
+}
+
+func (s *ReferralAPISuite) SetupTest() {
+	s.clearTables()
```
Comment (gordonBusyman) [inline tests/main_test.go:58]: Why it does the same with `TearDownTest()`?

**Review by gordonBusyman — CHANGES_REQUESTED** [review]
Description of the PR is missing

File: tests/main_test.go (hunk 1 above)
Comment (cyberdr0id) [reply tests/main_test.go:58]: To always work with an empty database before testing and exclude the situation of storing data in the database after

**Review by gordonBusyman — APPROVED** [review]

File: tests/database_test.go [hunk 2]
```diff
@@ -18,3 +30,109 @@ func (s *ReferralAPISuite) TestCreateUser() {
...
+	if err != nil {
+		s.FailNow(fmt.Errorf("cannot create user: %w", err).Error())
+	}
+	s.NoError(err)
+
+	_, err = s.repo.AddCandidate(id, defaultCandidateName, defaultCandidateSurname, defaultFileID)
+	if err != nil {
+		s.FailNow(fmt.Errorf("cannot add candidate: %w", err).Error())
+	}
+	s.NoError(err)
```
Comment (gordonBusyman) [inline tests/database_test.go:81]: You can probably move this to a separate function as it called several times here

**Review by gordonBusyman — APPROVED** [review]
One comment, not blocking

File: tests/database_test.go (hunk 2 above)
Comment (cyberdr0id) [reply tests/database_test.go:81]: I thought about it, but I decide to make like this, cause I thought that it's bad practice in tests to move code that called several times to a separate function and I thought that it's minor inaccuracy

Comment (gordonBusyman) [reply tests/database_test.go:81]: That's ok, not critical 👍

**Pull request merged**